http/src/main/java/io/hyperfoil/http/connection/Http1xConnection.java

```diff
@@ -159,8 +159,12 @@ public void request(HttpRequest request,
          buf.writeByte('\r').writeByte('\n');
       }
 
-      HttpCache httpCache = HttpCache.get(request.session);
-      httpCache.beforeRequestHeaders(request);
+      HttpCache httpCache = null;
+      if (request.hasCacheControl()) {
```

Comment:
it is still fine to have this method, I like it - just mask it with a null check